### PR TITLE
feat/doc UI cmpt improv

### PR DESCRIPTION
**Pull request by astagnol**
**Storybook "UI Components" category rework:**

**Global:**
- renamed the category to "ODS Components"
- renamed "introduction" page to "overview" and disabled it for now(generated overview should be done in a separate task)
- removed Notes, Sandbox and Docs tabs
- add sections to group our components: Actions, Content, Form, Layout, Navigation, User feedback
- removed table-of-contents.mdx & get-link.mdx in storybook/core
- readme in dist for each component is not generated anymore
- reworked specifications/usage for several components when necessary

**For each component:**
- removed contents, tests mdx packages/components/%component%/documentation/specifications/%component%
- removed contents/tests imports in packages/components/%component%/documentation/specifications/%component%/specifications-%component%.mdx
- replaced in component stories "UI Components" with "ODS Components" for title
- renamed "Web Component" category to "Demo"
- replaced "web-components" with "demo" for %component%.web-components.stories.ts  
- added Design Specifications with a link to zeroheight in Specifications category
- add Usage Guidelines category: moved Docs tab content here (usage.mdx)
- removed parameters (notes, docs), changelog and page imports in %component%.web-components.stories.ts
- removed notes at the bottom in Specifications when relevant
- ordered stories so we have: Specifications, Usage Guidelines and Demo

### CODE REVIEW

File: packages/storybook/.storybook/preview.js [hunk 1]
```diff
@@ -29,12 +29,15 @@ export const parameters = {
   previewTabs: {
     'storybook/docs/panel': {
       title: 'Docs',
+      hidden: true
```
Comment (dpellier) [inline packages/storybook/.storybook/preview.js:32]: can't you just remove them ?

**Review by dpellier — CHANGES_REQUESTED** [review]
Why are you naming the file in each stories `1_<compo>.specifications.stories.mdx`?
Files are already in a compo dir, you can just have specifications.stories.mdx on each

WDYT about having all the component categories open on first load?
Would be easier for the user to get a quick overview of all the list

**Review by aesteves60 — APPROVED** [review]

File: packages/storybook/.storybook/manager.js [hunk 2]
```diff
@@ -37,14 +43,39 @@ document.addEventListener('DOMContentLoaded', function() {
             }
             selector.parentNode.insertBefore(element, selector.nextSibling);
             observer.disconnect();
+            releaseSelectorConfig.enabled = false;
```
Comment (dpellier) [inline packages/storybook/.storybook/manager.js:46]: `disconnect` should be enough, you get the observer as second args on the callback function.
```
const callback = function(mutationsList, observer) {
  ...
  observer.disconnect();
}
```

File: packages/storybook/.storybook/manager.js [hunk 3]
```diff
@@ -37,14 +43,39 @@ document.addEventListener('DOMContentLoaded', function() {
...
+  // trigger a click on ODS Components sections
+  const observeItems = () => {
+    const callback = function(mutationsList) {
+      for (let i = 0, len = mutationsList.length; i < len; i++) {
+        if (mutationsList[i].type === 'childList' && itemsObserverConfig.enabled) {
+          let items = Array.from(document.querySelectorAll(".sidebar-item[id*='ods-components-'][data-nodetype='group']"));
+          if (items.length > 0) {
+            items.forEach(item => item.click());
+            observer.disconnect();
+            itemsObserverConfig.enabled = false;
```
Comment (dpellier) [inline packages/storybook/.storybook/manager.js:65]: same

**Review by dpellier — CHANGES_REQUESTED** [review]
I can't click on the version select
Capture d’écran 2023-11-06 à 17 14 36

File: packages/storybook/.storybook/manager.js [hunk 4]
```diff
@@ -62,7 +61,6 @@ document.addEventListener('DOMContentLoaded', function() {
           if (items.length > 0) {
             items.forEach(item => item.click());
             observer.disconnect();
-            itemsObserverConfig.enabled = false;
```
Comment (dpellier) [inline packages/storybook/.storybook/manager.js]: then remove also the const declaration and their usage in the if

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by dpellier — APPROVED** [review]

**Pull request merged**